server/tso/tso.go

```diff
@@ -234,8 +234,8 @@ func (t *timestampOracle) resetUserTimestamp(leadership *election.Leadership, ts
 	if err != nil {
 		return err
 	}
-	// save into etcd only if the time difference is big enough
-	if typeutil.SubTimeByWallClock(nextPhysical, t.tsoMux.tso.physical) > 3*updateTimestampGuard {
+	// save into etcd only if nextPhysical is close to lastSavedTime
+	if typeutil.SubTimeByWallClock(t.lastSavedTime.Load().(time.Time), nextPhysical) <= updateTimestampGuard {
```

Comment:
Why remove 3 here？

Reply:
The point is not removing 3 here. The old way compares `nextPhysical` with TSO in memory wrongly. We should compare `nextPhysical` with `lastSavedTime` to determine whether save into etcd or not.